mcs/class/corlib/Test/System.Threading/ThreadTest.cs

```diff
@@ -908,6 +909,25 @@ public void CurrentThread_Domains ()
 		}
 #endif // MONO_FEATURE_MULTIPLE_APPDOMAINS
 
+		[Test]
+		public void SetNameInThreadPoolThread ()
+		{
+			for (int i = 0; i < 10; ++i) {
```

Comment:
You want to run on multiple threading threads, and multiple times on the same one.